pretty_gpx/common/drawing/elevation_stats_section.py

```diff
@@ -14,6 +14,9 @@
 
 def downsample(x: np.ndarray, y: np.ndarray, n: int) -> tuple[np.ndarray, np.ndarray]:
     """Downsample the signla Y evaluate at X to N points, applying a simple moving average smoothing beforehand."""
+    if len(x) <= n:
+        assert len(x) == len(y)
```

Comment:
this check must always be true
you can use `assert_same_len((x,y))` at the beginning of the method

`from pretty_gpx.common.utils.asserts import assert_same_len`

Reply:
@ThomasParistech fixed